docs/linalg.rst

```diff
@@ -17,4 +17,145 @@ Matrices
 - ``R, S = ti.polar_decompose(A, ti.f32)``
 - ``U, sigma, V = ti.svd(A, ti.f32)`` (Note that ``sigma`` is a ``3x3`` diagonal matrix)
 
-TODO: doc here better like Vector.
+TODO: doc here better like Vector. WIP
+
+A matrix in Taichi can have two forms:
+
+  - as a temporary local variable. An ``n by m`` matrix consists of ``n * m`` scalar values.
+  - as a an element of a global tensor. In this case, the tensor is an ND array of ``n by m`` matrices.
+
+Declaration
+-----------
+
+As global tensors of matrices
+++++++++++++++++++++++++++++
+
+.. function:: ti.Matrix(n, m, dt=type, shape=shape)
+
+    :parameter n: (scalar) the number of rows in the matrix
+    :parameter m: (scalar) the number of columns in the matrix
+    :parameter type: (DataType) data type of the components
+    :parameter shape: (scalar or tuple) shape the tensor of vectors, see :ref:`tensor`
+
+    For example, this creates a 5x4 tensor of 3x3 matrices:
+    ::
+
+        # Python-scope
+        a = ti.Matrix(3, 3, dt=ti.f32, shape=(5, 4))
+
+.. note::
+
+    In Python-scope, ``ti.var`` declares :ref:`scalar_tensor`, while ``ti.Matrix`` declares tensors of matrices.
+
+
+As a temporary local variable
++++++++++++++++++++++++++++++
+
+.. function:: ti.Matrix([x, y, ...])
+
+    :parameter x: (scalar) the first component of the vector
+    :parameter y: (scalar) the second component of the vector
+
+    For example, this creates a 3x1 matrix with components (2, 3, 4):
+    ::
+
+        # Taichi-scope
+        a = ti.Matrix([2, 3, 4])
+
+.. note::
+
+    this is equivalent to ti.Vector([x, y, ...])
+
+
+.. function:: ti.Matrix([[x, y, ...], [z, w, ...], ...])
+
+    :parameter x: (scalar) the first component of the first column
```

Comment:
Actually, by default, `[x, y]` would be the elements of the first **row** :-)

This makes `[[x, y], [z, w]]` consistent with `list` of lists, `numpy` ndarrays and math notations $M_{i,j}$: the first index is for the row, the second the column.